src/atoms/CustomHtml/CustomHtml.js

```diff
@@ -1,20 +1,67 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import whitelist from '../../utils/whitelist.json';
 
-const CustomHtml = (props) => {
-  if (props.minimal) {
-    return null;
+class CustomHtml extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      html: ''
+    };
   }
-  const markup = { __html: props.nodeData.attrs.html };
-  return <div dangerouslySetInnerHTML={markup} />;
-};
 
-CustomHtml.propTypes = {
-  nodeData: PropTypes.object
-};
+  componentDidMount() {
+    this.setState({
+      html: cleanHtml(this.props.nodeData.attrs)
+    });
+  }
+
+  render() {
+    if (this.props.minimal) {
+      return null;
+    }
+
+    const markup = { __html: this.state.html };
+
+    return <div className="customHtml" dangerouslySetInnerHTML={markup} />;
+  }
+}
 
 CustomHtml.propTypes = {
   nodeData: PropTypes.object,
   minimal: PropTypes.bool
 };
+
 export default CustomHtml;
+
+function cleanHtml(attrs) {
+  let doc = new DOMParser().parseFromString(attrs.html, 'text/html');
+
+  // see if there are any scripts or iframes in the code
+  let iframes = doc.querySelectorAll('iframe');
+  let scripts = doc.querySelectorAll('script');
+  let fallbackUrl = attrs.fallback_url;
+  let whitelistRegex = new RegExp(whitelist.approved.join('|'));
+
+  let cond = {
+    isIframe: iframes.length > 0,
+    hasScript: scripts.length > 0,
+    hasFallbackUrl: fallbackUrl || fallbackUrl !== '' || fallbackUrl !== null,
+    isApproved: whitelistRegex.test(fallbackUrl) // TODO: test me in isolation
+  };
+
+  if (!cond.isIframe && cond.hasScript) {
+    // if a valid script is provided but not in an iframe, wrap in an iframe
+    if (cond.hasFallbackUrl && cond.isApproved) {
+      return `<iframe style="over" width="100%" height="650px" frameborder="0" scrolling="yes" marginheight="0" marginwidth="0" src="${fallbackUrl}"></iframe>`;
```

Comment:
I might be wrong, but I don't think the `style="over"` will do anything here

Reply:
It absolutely won't, and I don't know what that is or how it got in there. Good catch!